Source/Core/Core/MemoryWatcher.h

```diff
@@ -5,10 +5,16 @@
 #pragma once
 
 #include <map>
+#ifdef _WIN32
+#include <Windows.h>
```

Comment:
I'm always a little concerned when someone includes (or rather, has to) Windows.h in a header file; they tend to mess up a lot of things (most prominently the `min`/`max` defines and some other stuff), so can you try to avoid that?

What do you need it in here for specifically, `HANDLE` and `DWORD` (or do you get those from `CommonTypes.h`)? Either way, why do you have to include _both_ here; if you need them in the cpp file please include them there (instead of here in the header).

Reply:
Any Windows-based project should always define `NOMINMAX` globally. This disables the internal min/max macros from `Windows.h`.

Reply:
One method of isolation would be something like the [PImpl](http://en.cppreference.com/w/cpp/language/pimpl) idiom, which keeps all OS-related header dependencies fully in the cpp file in general, so no OS at all needs to 'leak' headers.

Reply:
Windows.h was included for `CreateFile()` and `WriteFile()`. I had included it in the header since the headers for managing pipes on linux (`sys/socket.h` and `sys/un.h`) were included there as well.

In the latest commit I changed it to `fileapi.h` for `CreateFile()` and `WriteFile()` and `handleapi.h` for `CloseHandle()`.